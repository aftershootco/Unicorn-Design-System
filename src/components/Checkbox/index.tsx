import clsx from 'clsx'
import React, { useCallback, useState } from 'react'
import styles from './styles.module.scss'

export interface CheckboxProps {
	defaultValue?: boolean
	onChange: (e: boolean) => void
}

const Checkbox: React.FC<CheckboxProps> = ({ defaultValue = false, onChange }) => {
	const [state, setState] = useState(defaultValue)
	const onClick = useCallback(
		() =>
			setState((state) => {
				onChange(!state)
				return !state
			}),
		[setState, onChange]
	)

	return (
		<div className={clsx(styles.checkbox, { [styles.active]: state })} onClick={onClick}>
			<span className={styles.checkmark}></span>
		</div>
	)
}

export default Checkbox
