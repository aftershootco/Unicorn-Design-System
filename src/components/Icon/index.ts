import ClearIcon from './ClearIcon'
import ExpandMore from './ExpandMore'
import ContactSupportIcon from './ContactSupport'
import CheckCircle from './CheckCircle'
import Check from './Check'
import ArrowDownRounded from './ArrowDownRounded'
import ArrrowUpRounded from './ArrrowUpRounded'
import CloseRounded from './CloseRounded'
import ContentCopy from './ContentCopy'
import StarRate from './StarRate'
import ArrowForwardRounded from './ArrowForwardRounded'
import ChevronLeftIcon from './ChevronLeftIcon'
import KeyBoardArrowDown from './KeyBoardArrowDown'
import KeyBoardArrowUp from './KeyBoardArrowUp'
import Menu from './Menu'
import ChevronRightIcon from './ChevronRightIcon'
import KeyBoardLeftIcon from './KeyBoardLeftIcon'
import KeyBoardRightIcon from './KeyBoardRightIcon'

export {
	ClearIcon,
	ExpandMore,
	ContactSupportIcon,
	CheckCircle,
	Check,
	ArrowDownRounded,
	ArrrowUpRounded,
	CloseRounded,
	ContentCopy,
	StarRate,
	ArrowForwardRounded,
	ChevronLeftIcon,
	KeyBoardArrowDown,
	KeyBoardArrowUp,
	Menu,
	ChevronRightIcon,
	KeyBoardLeftIcon,
	KeyBoardRightIcon,
}
