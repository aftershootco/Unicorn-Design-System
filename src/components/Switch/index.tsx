import clsx from "clsx";
import React, { useCallback, useState } from "react";
import styles from "./styles.module.scss";

export interface SwitchProps {
  defaultValue?: boolean;
  onChange: (e: boolean) => void;
}

const Switch: React.FC<SwitchProps> = ({ defaultValue = false, onChange }) => {
  const [state, setState] = useState(defaultValue);
  const onClick = useCallback(() => {
    setState((state) => {
      onChange(!state);
      return !state;
    });
  }, [onChange, setState]);

  return (
    <label
      className={clsx(styles.switch, { [styles.active]: state })}
      onClick={onClick}
    >
      <span className={clsx(styles.round, styles.slider)}></span>
    </label>
  );
};

export default Switch;
