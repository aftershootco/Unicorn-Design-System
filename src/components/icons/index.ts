import ClearIcon from './ClearIcon'
import ExpandMore from './ExpandMore'
import ContactSupportIcon from './ContactSupport'
import CheckCircle from './CheckCircle'
import Check from './Check'
import ArrowDownRounded from './ArrowDownRounded'
import ArrrowUpRounded from './ArrrowUpRounded'
import CloseRounded from './CloseRounded'
import ContentCopy from './ContentCopy'
import StarRate from './StarRate'
import ArrowForwardRounded from './ArrowForwardRounded'
import ChevronLeftIcon from './ChevronLeftIcon'

export {
	ClearIcon,
	ExpandMore,
	ContactSupportIcon,
	CheckCircle,
	Check,
	ArrowDownRounded,
	ArrrowUpRounded,
	CloseRounded,
	ContentCopy,
	StarRate,
	ArrowForwardRounded,
	ChevronLeftIcon,
}
