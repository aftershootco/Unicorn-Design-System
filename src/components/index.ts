import '../index.scss'
import Accordion, { AccordionProps } from './Accordion/Accordion'
import Button, { ButtonProps } from './Button/Button'
import Carousel, { CarouselProps } from './Carousel/Carousel'
import CheckBox, { CheckBoxProps } from './CheckBox/CheckBox'
import Chip, { ChipProps } from './Chip/Chip'
import Divider, { DividerProps } from './Divider/Divider'
import DropDown, { DropdownData, DropDownProps } from './DropDown/DropDown'
import LinearProgressBar, { LinearProgressBarProps } from './LinearProgressBar/LinearProgressBar'
import RadioButtons from './RadioButton/RadioButton'
import Slider from './Slider/Slider'
import Switch, { SwitchProps } from './Switch/Switch'
import Tabs, { TabsProps } from './Tabs/Tabs'
import TextInput, { TextInputProps } from './TextInput/TextInput'
import Toggle, { ToggleProps } from './Toggle/Toggle'
import ToolTip, { ToolTipProps } from './ToolTip/ToolTip'
import HeadlessDropdown, { HeadlessDropdownData, HeadlessDropdownProps } from './HeadlessDropdown/HeadlessDropdown'
import NewDropDown, { NewDropDownProps, NewDropdownData } from './NewDropDown/NewDropDown'

export {
	Accordion,
	AccordionProps,
	Button,
	ButtonProps,
	Carousel,
	CarouselProps,
	Slider,
	DropDown,
	DropdownData,
	DropDownProps,
	ToolTip,
	ToolTipProps,
	Tabs,
	TabsProps,
	CheckBox,
	CheckBoxProps,
	Chip,
	ChipProps,
	Divider,
	DividerProps,
	LinearProgressBar,
	LinearProgressBarProps,
	RadioButtons,
	Switch,
	SwitchProps,
	TextInput,
	TextInputProps,
	Toggle,
	ToggleProps,
	HeadlessDropdown,
	HeadlessDropdownData,
	HeadlessDropdownProps,
	NewDropDown,
	NewDropDownProps,
	NewDropdownData,
}
