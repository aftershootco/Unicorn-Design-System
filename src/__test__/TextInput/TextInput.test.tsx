import React from "react";
import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import { TextInput } from "../../components";

describe(`Testing -> TextInput`, () => {
    it(`to get snapshot`, () => {
		const tree = renderer.create(<TextInput value='aftershoot'/>).toJSON()
		expect(tree).toMatchSnapshot()
	})

    it(`should be in the document`, () => {
        render(<TextInput value='aftershoot'/>);
        expect(screen.getByTestId('test1')).toBeInTheDocument();
    })

    const temp1:('primary' | 'secondary' | 'tertiary')[] = ['primary','secondary','tertiary'];
    for(let i=0; i<temp1.length; ++i){
        it(`should display the required div according to variant`, () => {
            render(<TextInput value='aftershoot' variant={temp1[i]}/>);
            if(temp1[i] === 'primary' || temp1[i] === 'secondary'){
                expect(screen.getByTestId('test1')).toBeInTheDocument();
            }else{
                expect(screen.getByTestId('test2')).toBeInTheDocument();
            }
        })
    }

    it(`should display the required div according to variant:-primary and type:-password`, () => {
        render(<TextInput value='aftershoot' variant='primary' type='password' />)
        expect(screen.getByTestId('test1')).toBeInTheDocument();
        expect(screen.getByTestId('test3')).toBeInTheDocument();
    })

    it(`should display the required div according to variant:-primary and type:-text`, () => {
        render(<TextInput value='aftershoot' variant='primary' type='text' />)
        expect(screen.getByTestId('test1')).toBeInTheDocument();
    })

    it(`should display the required div according to variant:-tertiary and type:-text`, () => {
        render(<TextInput value='aftershoot' variant='tertiary' type='text' />)
        expect(screen.getByTestId('test2')).toBeInTheDocument();
    })
})