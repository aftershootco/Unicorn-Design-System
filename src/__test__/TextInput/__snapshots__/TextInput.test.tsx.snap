// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing -> TextInput to get snapshot 1`] = `
<div
  className="relative undefined"
>
  <input
    className="false inputBox undefined"
    data-testid="test1"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="text"
    value="aftershoot"
  />
</div>
`;
